src/main/java/com/salesforce/functions/jvm/sdk/data/bulk/BulkApi.java

```diff
@@ -0,0 +1,276 @@
+
+  /**
+   * Aborts a query job.
+   *
+   * @param reference The reference of the job to abort.
+   * @throws DataApiException If an API error occurred during the operation.
+   * @see <a
+   *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/query_abort_job.htm">API
+   *     Reference</a>
+   */
+  @SuppressWarnings("unused")
+  void abortJob(QueryJobReference reference) throws DataApiException;
+
+  /**
+   * Deletes an ingest job.
+   *
+   * @param reference The reference of the job to delete.
+   * @throws DataApiException If an API error occurred during the operation.
+   * @see <a
+   *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/delete_job.htm">API
+   *     Reference</a>
+   */
+  @SuppressWarnings("unused")
+  void deleteJob(IngestJobReference reference) throws DataApiException;
+
+  /**
+   * Deletes a query job.
+   *
+   * @param reference The reference of the job to delete.
+   * @throws DataApiException If an API error occurred during the operation.
+   * @see <a
+   *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/query_delete_job.htm">API
+   *     Reference</a>
+   */
+  @SuppressWarnings("unused")
+  void deleteJob(QueryJobReference reference) throws DataApiException;
+
+  /**
+   * Retrieves a list of successfully processed records for a completed ingest job.
+   *
+   * @param reference The reference of the job to retrieve the list for.
+   * @return A {@link DataTable} with the original uploaded data augmented with 'sf__Created' and
+   *     'sf__Id' fields.
+   * @throws DataApiException If an API error occurred during the operation.
+   * @see <a
+   *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_successful_results.htm">API
+   *     Reference</a>
+   * @see BulkApi#getFailedRecordResults(IngestJobReference)
+   * @see BulkApi#getUnprocessedRecordResults(IngestJobReference)
+   */
+  @Nonnull
+  @SuppressWarnings("unused")
+  DataTable getSuccessfulRecordResults(IngestJobReference reference) throws DataApiException;
+
+  /**
+   * Retrieves a list of failed records for a completed ingest job.
+   *
+   * @param reference The reference of the job to retrieve the list for.
+   * @return A {@link DataTable} with the original uploaded data augmented with 'sf__Error' and
+   *     'sf__Id' fields.
+   * @throws DataApiException If an API error occurred during the operation.
+   * @see <a
+   *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_failed_results.htm">API
+   *     Reference</a>
+   * @see BulkApi#getSuccessfulRecordResults(IngestJobReference)
+   * @see BulkApi#getUnprocessedRecordResults(IngestJobReference)
+   */
+  @Nonnull
+  @SuppressWarnings("unused")
+  DataTable getFailedRecordResults(IngestJobReference reference) throws DataApiException;
```

Comment:
i'd like to see this be a specialization of `DataTable` that exposes the accessors for:

* `sf__Error: string`
* `sf__Id: string`

e.g.; `FailedResultsDataTable`?